### PR TITLE
[CryptoMiniSat] Add Builder

### PR DESCRIPTION
**Pull request by Wimmerer**
There's a weird issue I can't really figure. On musl I get `/workspace/srcdir/cryptominisat/src/main_exe.cpp:34:5: error: ‘feenableexcept’ was not declared in this scope`. But if I inspect that section of the code I see it's in an `#if defined(__GNUC__) && defined(__linux__)` block. That seems like it shouldn't be defined on musl then to me?

### CODE REVIEW

**Comment by giordano** [comment]
>  But if I inspect that section of the code I see it's in an #if defined(__GNUC__) && defined(__linux__) block. That seems like it shouldn't be defined on musl then to me?

[`__GNUC__`](https://gcc.gnu.org/onlinedocs/cpp/Common-Predefined-Macros.html) is

> defined by all GNU compilers that use the C preprocessor

It wouldn't be surprising if they have a wrong guard and that should be something like `__GLIBC__` instead

**Comment by Wimmerer** [comment]
Ok thanks @giordano I'll ask upstream. Do you have any idea what I'm doing wrong about Boost on Windows? I tried setting all sorts of specific paths for CMake, and nothing worked.

**Comment by giordano** [comment]
> Do you have any idea what I'm doing wrong about Boost on Windows?

Apart from the fact that CMake can't do arithmetic?
```
[15:53:10] -- Could NOT find Boost: missing: program_options (found /opt/x86_64-w64-mingw32/x86_64-w64-mingw32/sys-root/usr/local/lib/cmake/Boost-1.71.0/BoostConfig.cmake (found suitable version "1.71.0", minimum required is "1.46"))
```

**Comment by Wimmerer** [comment]
It's because it's got a patch number while the required doesn't I bet?

File: C/CryptoMiniSat/build_tarballs.jl [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+# These are the platforms we will build for by default, unless further
+# platforms are passed in on the command line
+platforms = expand_cxxstring_abis(supported_platforms())
+
+# The products that we will ensure are always built
+products = Product[
+    ExecutableProduct("cryptominisat5", :cryptominisat5),
+    ExecutableProduct("cryptominisat5_simple", :cryptominisat5_simple),
+    LibraryProduct("libcryptominisat5", :libcryptominisat5),
```
Comment (giordano) [inline C/CryptoMiniSat/build_tarballs.jl:40]:
```suggestion
    LibraryProduct(["libcryptominisat5", "libcryptominisat5win"], :libcryptominisat5),
```

File: C/CryptoMiniSat/build_tarballs.jl [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+dependencies = [
+    Dependency("boost_jll"),
+    Dependency("Zlib_jll"),
+    Dependency("SQLite_jll"),
+    Dependency("MPICH_jll"),
+    Dependency("MicrosoftMPI_jll")
+]
+
+# Build the tarballs, and possibly a `build.jl` as well.
+build_tarballs(ARGS, name, version, sources, script, platforms, products, dependencies)
```
Comment (giordano) [inline C/CryptoMiniSat/build_tarballs.jl:54]:
```suggestion
build_tarballs(ARGS, name, version, sources, script, platforms, products, dependencies; preferred_gcc_version=v"7")
```

File: C/CryptoMiniSat/build_tarballs.jl [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+install_license ${WORKSPACE}/srcdir/cryptominisat/LICENSE.txt
+"""
+
+# These are the platforms we will build for by default, unless further
+# platforms are passed in on the command line
+platforms = expand_cxxstring_abis(supported_platforms())
+
+# The products that we will ensure are always built
+products = Product[
+    ExecutableProduct("cryptominisat5", :cryptominisat5),
```
Comment (giordano) [inline C/CryptoMiniSat/build_tarballs.jl:38]:
```suggestion
    ExecutableProduct(["cryptominisat5", "cryptominisat5win"], :cryptominisat5),
```

**Comment by Wimmerer** [comment]
I'm going to remove MicrosoftMPI_jll, it wasn't sucessfully built against, and unless someone specifically asks for it I *feel* like we're safe without it?

**Pull request merged**